### PR TITLE
Refactor platform tip handling

### PR DESCRIPTION
**Pull request by znarf**

### CODE REVIEW

File: server/paymentProviders/opencollective/collective.js [hunk 1]
```diff
@@ -53,42 +54,45 @@ paymentMethodProvider.processOrder = async order => {
...
+    hostCurrencyFxRate,
     amountInHostCurrency,
     hostFeeInHostCurrency,
-    platformFeeInHostCurrency,
     taxAmount: order.taxAmount,
-    paymentProcessorFeeInHostCurrency: 0,
     description: order.description,
     data: {
       isFeesOnTop: order.data?.isFeesOnTop,
+      hasPlatformTip: platformTip ? true : false,
```
Comment (Betree) [inline server/paymentProviders/opencollective/collective.js:91]: Maybe we have an opportunity to solve https://github.com/opencollective/opencollective/issues/4236 here? By adding an additional flag like `isElligibleToPlatformTip`

File: server/paymentProviders/stripe/creditcard.js [hunk 2]
```diff
@@ -197,22 +196,31 @@ const createChargeAndTransactions = async (hostStripeAccount, { order, hostStrip
   const hostFee = await getHostFee(order, host);
   const hostFeeInHostCurrency = Math.round(hostFee * hostCurrencyFxRate);
 
+  const fees = extractFees(balanceTransaction, balanceTransaction.currency);
+
+  let platformTipInHostCurrency, platformFeeInHostCurrency;
+  const platformTip = getPlatformTip(order);
+  if (platformTip) {
+    platformTipInHostCurrency = isSharedRevenue ? platformTip * hostCurrencyFxRate || 0 : fees.applicationFee;
```
Comment (Betree) [inline server/paymentProviders/stripe/creditcard.js:204]: It was there from before, but looks to me like this one should be rounded

```suggestion
    platformTipInHostCurrency = isSharedRevenue ? Math.round(platformTip * hostCurrencyFxRate) || 0 : fees.applicationFee;
```

**Review by Betree — APPROVED** [review]
The code is much clearer like that :ok_hand:

File: server/paymentProviders/opencollective/collective.js (hunk 1 above)
Comment (znarf) [reply server/paymentProviders/opencollective/collective.js:91]: I added support for this but no migration for past data at this point.

File: server/paymentProviders/stripe/creditcard.js (hunk 2 above)
Comment (znarf) [reply server/paymentProviders/stripe/creditcard.js]: Added it back.

Comment (Betree) [reply server/paymentProviders/stripe/creditcard.js]: Sorry, my feedback was not very clear: the `Math.round` was not there in the previous version. Anyway, I think it's a good addition.

**Pull request merged**